### ISSUE TITLE
Can't give a user more than one role in a project.

### ISSUE THREAD

**Issue by mymusise** [doccano/doccano#537]
When I given a user two role(project_admin and annotator), I got a error with:
```
backend_1   |   File "/app/api/serializers.py", line 98, in get_current_users_role
backend_1   |     queryset, project=instance.id, user=self.context.get("request").user.id
backend_1   |   File "/usr/local/lib/python3.6/site-packages/django/shortcuts.py", line 93, in get_object_or_404
backend_1   |     return queryset.get(*args, **kwargs)
backend_1   |   File "/usr/local/lib/python3.6/site-packages/django/db/models/query.py", line 403, in get
backend_1   |     (self.model._meta.object_name, num)
backend_1   | api.models.RoleMapping.MultipleObjectsReturned: get() returned more than one RoleMapping -- it returned 2!
```

There may more than one role of `users_role`, so it should replace `get_object_or_404` with `filter` :

[https://github.com/doccano/doccano/blob/v1.0.2/app/api/serializers.py#L98](url)

```python
        queryset = RoleMapping.objects.values("role_id__name")
        if queryset:
            users_role = get_object_or_404(
                queryset, project=instance.id, user=self.context.get("request").user.id
            )
            for key, val in role_abstractor.items():
                role_abstractor[key] = users_role["role_id__name"] == val
        return role_abstractor
```

**Comment by c-w** [comment]
The roles as introduced in https://github.com/doccano/doccano/pull/333 were designed to be incremental (see table below):

View ↓ / Role →  | Project Admin  | Annotator | Annotation Approver |
---------------- |  ------------- | --------- | ------------------- |
Me               | x              | x         | x                   |
Features         | x              |x          | x                   |
ProjectList      | x              |x          | x                   | 
ProjectDetail    | x              |x          | x                   |
StatisticsAPI    | x              |x          | x                   |
ApproveLabelsAPI | x              |           | x                   |
LabelList        | x              |x          | x                   |
LabelDetail      | x              |x          | x                   |
DocumentList     | x              |x          | x                   |
DocumentDetail   | x              |x          | x                   |
AnnotationList   | x              |x          | x                   |
AnnotationDetail | x              |x          | x                   |
TextUploadAPI    | x              |           |                     |
CloudUploadAPI   | x              |           |                     |
TextDownloadAPI  | x              |           |                     |

As such, it's by design that a user can only have one role on a project. For example, it wouldn't make sense for a user to have both "annotator" and "project admin" roles since the latter grants all the same permissions as the former and then some more.

Could you clarify what is the use-case for granting a user two or more roles on the same project?

**Comment by kuraga** [comment]
Hm... But UI allows this.

**Comment by c-w** [comment]
@kuraga Is your observation for the new [v1 UI](https://raw.githubusercontent.com/doccano/doccano/master/docs/demo.gif) or the old [v0 UI](https://doccano.herokuapp.com/demo/text-classification/)?

**Comment by kuraga** [comment]
@c-w v1. I have two different `a_user`'s roles on page `/projects/1/members`. (BTW, I get a 50x error on attempt to add the same role for the same user.)

**Comment by mymusise** [comment]
@c-w  Thanks, so, it should be forbidden to create a new role when one exists on a project, update only?

**Comment by c-w** [comment]
@mymusise Yes, that's correct. Each user/project/role pair should be unique. If this is not currently being enforced, I suspect it's a bug with the v1 UI.

@kuraga That sounds like a bug in the validation logic in [RoleMapping](https://github.com/doccano/doccano/blob/65e80b89e2cb2de6e7963d70cf0a307c4481ea60/app/api/models.py#L255-L269). The error for that situation should be a 400.

**Comment by kuraga** [comment]
Let's mark as `bug` instead of `enhancement` since we have an error (server exception).

**Comment by Charlyo** [comment]
Had this issue happen to me. Is there any way to fix this? (removing one role)? 
I do have al the labeling done and not project appears / cannot access members page.

**Comment by ljades** [comment]
Is there an update on this? It seemed like consensus was reached to mark this as a bug instead of a feature, correct? It should not be allowed to add a user to more than one role.

It seems like any user who is currently assigned to a role for a given project already should not show up in the dropdown (or be validated against if selected) for new project member roles.
